create metadata files for all connectors

alafanechere opened the pull request:
## What
Closes https://github.com/airbytehq/airbyte/issues/24848
I ran the `persist_metadata_definition` Dagster asset to generate these files.

evantahler reviewed (commented):
It's Happening 🥳

bnchrch commented:
@alafanechere Before we merge lets make sure this PR is in
https://github.com/airbytehq/airbyte/pull/24840

bnchrch reviewed (changes requested):
@alafanechere Looking good!

I think we should do the following  adjustments
1. Move any non airbyte connectors under `connectors/third-party/<AUTHOR>/<CONNECTOR>`. That should currently apply to `farosai` and `ghcr.io`
2. Remove the `devmate-cloud` subfolder of `ghcr.io`

bnchrch commented on airbyte-integrations/connectors/source-mongodb-v2/metadata.yaml:
```diff
@@ -0,0 +1,20 @@
+  catalogs:
+    cloud:
+      dockerImageTag: 0.1.7
+      dockerRepository: airbyte/source-mongodb-strict-encrypt
```
@alafanechere Also I think for strict encrypt we need to manually add a metadata.yml file for each.

These metadata files should
1. Have both cloud and oss catalogs disabled
2. Contain enough information to build the correct docker image
3. Have a comment acknowledging that the get deployed by their non strict encrypt sibling

alafanechere commented:
@bnchrch I implemented your suggestions:
> Move any non airbyte connectors under connectors/third-party/<AUTHOR>/<CONNECTOR>. That should currently apply to farosai and ghcr.io

✅ 

> Rename the ghcr.io folder to ghcr

✅ 
> Remove the devmate-cloud subfolder of ghcr.io

✅ - The `ghcr` directory is now empty, is this expected @bnchrch?

> I think for strict encrypt we need to manually add a metadata.yml file for each.

I added metadata.yaml files for strict encrypt variants. They are all disabled from cloud and oss.

bnchrch reviewed (approved):
> ✅ - The ghcr directory is now empty, is this expected @bnchrch?

@alafanechere Oh whoops! I couldve been more clear when I said

> Remove the devmate-cloud subfolder of ghcr.io

We had the following file path in the last commit
`connectors/ghcr.io/devmate-cloud/source-something/metadata.yml`

And I was thinking it should have been
`connectors/third-party/ghcr/source-something/metadata.yml`


But thats minor so ill approve now butso when your back online if you could correct that then feel free to merge!

alafanechere commented:
@bnchrch I
- I re-added the deleted connector in `ghrc`
- I fixed the metadata not passing validation
- I updated the model to accept `message_queue` `connectorSubType` (do you think this should be done in a separate PR?)
- The workflow to run validation was quite bulky and sometime stuck on this PR. I tweaked it a bit to make it faster and reuse existing components [here](https://github.com/airbytehq/airbyte/pull/25107).

alafanechere commented:
> what led you to adding the subtype?

destination-mqtt uses the `message_queue` subtype

> Were we able to do this in the overrides in the dev assets?

I did this manually . I don't think we can do it in the override for all the cases:
* Missing release stages: we can fallback to alpha
* Missing subtype: We could fallback to unknown, but I think it's better to curate these  connector subtype properly now as there was only a few.

bnchrch commented:
> > what led you to adding the subtype?
> 
> destination-mqtt uses the `message_queue` subtype
> 
> > Were we able to do this in the overrides in the dev assets?
> 
> I did this manually . I don't think we can do it in the override for all the cases:
> 
> * Missing release stages: we can fallback to alpha
> * Missing subtype: We could fallback to unknown, but I think it's better to curate these  connector subtype properly now as there was only a few.

Hmm no big deal but if I understand correctly you should be able to override releaseStage and subType for any connector by its connectorId [here](https://github.com/airbytehq/airbyte/blob/augustin/metadata-service/upload-metadata-files-to-gcs/airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/dev.py#L136)

alafanechere commented:
The Dagger CI infra is currently down. I ran the metadata validation locally with success. Here are the [logs](https://gist.github.com/alafanechere/41c970ed9d5a99c15f5474e7be1ca7dd). Ready to merge.
